icu4c/source/common/uloc_keytype.cpp

```diff
@@ -325,12 +325,20 @@ initFromResourceBundle(UErrorCode& sts) {
             }
         }
         if (U_FAILURE(sts)) {
+            if (typeDataMap != NULL) {
+                uhash_close(typeDataMap);
+                typeDataMap = NULL;
+            }
             break;
         }
 
         LocExtKeyData* keyData = gLocExtKeyDataEntries->create();
         if (keyData == NULL) {
             sts = U_MEMORY_ALLOCATION_ERROR;
+            if (typeDataMap != NULL) {
```

Comment:
by using LocalUHashtablePointer as I stated above we won't need these if/uhash_close

Reply:
OK.